core/src/main/java/com/scalar/db/common/AbstractTwoPhaseCommitTransaction.java

```diff
@@ -59,8 +65,60 @@ public Optional<String> getTable() {
     return tableName;
   }
 
-  protected <T extends Mutation> List<T> copyAndSetTargetToIfNot(List<T> mutations) {
-    return ScalarDbUtils.copyAndSetTargetToIfNot(mutations, namespace, tableName);
+  @Override
+  public void mutate(List<? extends Mutation> mutations) throws CrudException {
+    checkArgument(!mutations.isEmpty(), CoreError.EMPTY_MUTATIONS_SPECIFIED.buildMessage());
+    for (Mutation mutation : mutations) {
+      if (mutation instanceof Put) {
+        put((Put) mutation);
+      } else if (mutation instanceof Delete) {
+        delete((Delete) mutation);
+      } else if (mutation instanceof Insert) {
+        insert((Insert) mutation);
+      } else if (mutation instanceof Upsert) {
+        upsert((Upsert) mutation);
+      } else {
+        assert mutation instanceof Update;
+        update((Update) mutation);
+      }
+    }
+  }
```

Comment:
Ditto. This is not directly related to this PR, but I also moved the `mutate()` method implementation to `AbstractTwoPhaseCommitTransaction`.